File: DSA/VaniJain_590012020/Day18Question2.java
import java.util.*;
public class InsertionSort 
{
    public static void main(String[] args) 
    {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        int n = scanner.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements:");
        for (int i = 0; i < n; i++) 
        {
            arr[i] = scanner.nextInt();
        }
        for (int i = 1; i < n; i++) 
        {
            int key = arr[i];      
            int j = i - 1;
            while (j >= 0 && arr[j] > key) 
            {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
        System.out.println("Sorted array in ascending order:");
        for (int i = 0; i < n; i++) 
        {
            System.out.print(arr[i] + " ");
        }
    }
}

File: DSA/Raima_Rekhi_590013023/Day18Question2.c
#include <stdio.h>

void insertion_sort(int arr[],int n);

int main(){
int n;

printf("Enter the size of the array:\n");
scanf("%d",&n);
int arr[n];

// Prompt the user to enter the elements of the array
printf("Enter the elements of the array:\n");
for(int i=0;i<n;++i){
scanf("%d",&arr[i]);
}

insertion_sort(arr,n);
printf("Sorted Array in ascending order:\n");
for(int i=0;i<n;++i){
printf("%d  ",arr[i]);
}
printf("\n");
return 0;
}

// Function to sort an array using insertion sort
void insertion_sort(int arr[],int n){
for(int i=1;i<n;++i){
int key=arr[i];
int j=i-1;
while(j>=0 && arr[j]>key){
arr[j+1]=arr[j];
j--;
}
arr[j+1]=key;
}
}
